diff --git a/src/surge-xt/gui/widgets/XMLConfiguredMenus.cpp b/src/surge-xt/gui/widgets/XMLConfiguredMenus.cpp
index ea3b546b..c4b6e31d 100644
--- a/src/surge-xt/gui/widgets/XMLConfiguredMenus.cpp
+++ b/src/surge-xt/gui/widgets/XMLConfiguredMenus.cpp
@@ -697,11 +697,14 @@ void FxMenu::populateForContext(bool isCalledInEffectChooser)
     if (sge)
     {
         cfxid = sge->effectChooser->currentClicked;
-        auto deactbm = sge->effectChooser->getDeactivatedBitmask();
-        addDeact = true;
-        isDeact = deactbm & (1 << cfxid);
-        cfxtype = sge->effectChooser->fxTypes[cfxid];
-        enableClear = cfxtype != fxt_off;
+        if (cfxid >= 0)
+        {
+            auto deactbm = sge->effectChooser->getDeactivatedBitmask();
+            addDeact = true;
+            isDeact = deactbm & (1 << cfxid);
+            cfxtype = sge->effectChooser->fxTypes[cfxid];
+            enableClear = cfxtype != fxt_off;
+        }
     }
 
     auto cfx = std::string{"Current FX Slot"};
